### DISCUSSION TITLE
Cost of embedding

### DISCUSSION THREAD

**Discussion by Amitt1412** [langfuse/langfuse#2479] [Support] [answered]
Hi, I have gone through the tutorials & docs. Upon integrating the langfuse with my langchain app, I can't see the cost of embedding, separately. I can't see the embedding model being used as well. Could someone help me with assistance on it?

Regards
Amit Tiwari

**Comment by marcklingen** [comment]
If you use langchain, embeddings are not included in the callbacks unfortunately as of now

**Comment by Amitt1412** [comment]
Since the same is important with respect to the cost per query. Does the overall cost per query include the same? 
Also, in the near future I would need to ingest a large amount of data & therefore It becomes critical for me to keep track of the cost calling the embedding models.

**Comment by marcklingen** [comment, marked as answer]
As mentioned above, langchain callbacks do not provide this. You could however ingest tokencounts of embedding models yourself, either by ingesting the token counts provided by an LLM API or by tokenizing the inputs yourself. You can ingest custom token counts into langfuse and this is frequently used to capture costs of embedding models, more on this here: https://langfuse.com/docs/model-usage-and-cost
